arangod/RocksDBEngine/RocksDBVPackIndex.cpp

```diff
@@ -1952,53 +2001,72 @@ Result RocksDBVPackIndex::remove(transaction::Methods& trx,
 
   size_t const count = elements.size();
 
-  if (_unique) {
-    for (size_t i = 0; i < count; ++i) {
-      // banish key in in-memory cache
-      invalidateCacheEntry(::lookupValueFromSlice(elements[i].string()));
-
+  for (size_t i = 0; i < count; ++i) {
+    if (_unique) {
       s = mthds->Delete(_cf, elements[i]);
-
-      if (!s.ok()) {
-        res.reset(rocksutils::convertStatus(s, rocksutils::index));
-      }
-    }
-    if (res.fail()) {
-      TRI_ASSERT(doc.get(StaticStrings::KeyString).isString());
-      addErrorMsg(res, doc.get(StaticStrings::KeyString).copyString());
+    } else {
+      // non-unique index contains the unique objectID written exactly once
+      s = mthds->SingleDelete(_cf, elements[i]);
     }
-  } else {
-    // non-unique index contain the unique objectID written exactly once
-    for (size_t i = 0; i < count; ++i) {
+
+    if (s.ok()) {
       // banish key in in-memory cache
       invalidateCacheEntry(::lookupValueFromSlice(elements[i].string()));
```

Comment:
Also need to call `trackIndexCacheRefill` here, resulting in the same duplicated code as above.

Reply:
addressed